src/sys/signal.rs

```diff
@@ -388,8 +388,12 @@ pub fn pthread_sigmask(how: SigFlags,
     Errno::result(res).map(drop)
 }
 
-pub fn kill(pid: libc::pid_t, signal: Signal) -> Result<()> {
-    let res = unsafe { libc::kill(pid, signal as libc::c_int) };
+pub fn kill(pid: libc::pid_t, signal: Option<Signal>) -> Result<()> {
```

Comment:
In any case, if making use of the new impl, we should include tests that demonstrate it works in both cases.